File: servers/expressjs/validation/BlogValidator.js
module.exports = function blogValidator(blog) {
  const errors = [];

  if (blog.title === '') {
    errors.push({
      field: 'title',
      message: 'Blog must contain a title.',
    });
  }

  if (blog.title.length > 120) {
    errors.push({
      field: 'title',
      message: 'Blog title cannot exceed 120 characters.',
    });
  }

  if (blog.body === '') {
    errors.push({
      field: 'body',
      message: 'Blog must contain a body.',
    });
  }

  return errors;
};

File: servers/dotnet/Application/Blogs/BlogValidator.cs
using Application.Blogs.DTOs;
using Application.Common;
using Application.Interfaces;

namespace Application.Blogs;

public class BlogValidator : IBlogValidator
{
    public List<ValidationError> Validate(CreateBlogDto blogDto)
    {
        List<ValidationError> errors = new List<ValidationError>();

        if (string.IsNullOrWhiteSpace(blogDto.Title))
        {
            errors.Add(new() { Field = "title", Message = "Blog must contain a title" });
        }
        if (string.IsNullOrWhiteSpace(blogDto.Body))
        {
            errors.Add(new() { Field = "body", Message = "Blog must contain a body" });
        }

        return errors;
    }
}
